crates/common/network_spec/src/networks.rs

```diff
@@ -16,6 +16,8 @@ pub enum Network {
 pub struct NetworkSpec {
     pub network: Network,
     pub genesis: Genesis,
+    pub chain_id: u64,
```

Comment:
```suggestion
```

Can you implement `chain_id()` as a function of the `Network` enum

So

```rust
impl Network {
    pub fn chain_id(&self) {
        match self {
            Network::Mainnet => 1,
            ... etc etc
        }
    }
} 
```

Reply:
Yeah